src/Transports/MassTransit.AmazonSqsTransport/AmazonSqsTransport/AmazonSqsReceiveLockContext.cs

```diff
@@ -15,16 +16,18 @@ public class AmazonSqsReceiveLockContext :
         static readonly TimeSpan MaxVisibilityTimeout = TimeSpan.FromHours(12);
         readonly CancellationTokenSource _activeTokenSource;
         readonly ClientContext _clientContext;
+        readonly Uri _inputAddress;
         readonly Message _message;
 
-        readonly ReceiveContext _receiveContext;
         readonly ReceiveSettings _settings;
+        readonly Stopwatch _timer;
         readonly Task _visibilityTask;
         bool _locked;
 
-        public AmazonSqsReceiveLockContext(ReceiveContext receiveContext, Message message, ReceiveSettings settings, ClientContext clientContext)
+        public AmazonSqsReceiveLockContext(Uri inputAddress, Message message, ReceiveSettings settings, ClientContext clientContext)
         {
-            _receiveContext = receiveContext;
+            _timer = Stopwatch.StartNew();
```

Comment:
This can just be a `DateTime` of when the lock was created, and simple date math below, vs using the Stopwatch.